.github/workflows/build.yml

```diff
@@ -1,19 +1,29 @@
-name: Test & Build
-
+name: build
 on:
+  release:
+    types: [created]
   push:
-    branches: '*'
-  pull_request:
-    branches: '*'
-
+    branches:
+      - '**'
 jobs:
-  test-build:
-    name: Run tests and build project
+  build:
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
         with:
-          node-version: 18
+          node-version: '20.x'
+          registry-url: 'https://registry.npmjs.org'
+          cache: npm
       - run: npm ci
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - run: npm run test:coverage
       - run: npm run build
+      - run: npm publish
+        if: github.event_name == 'release' && github.event.action == 'created'
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
I take it we need to add the `NPM_TOKEN` secret to this repository.
Is that generate from a specific account?

Reply:
Ah, it's an org secret, so should work out-of-the-box 👍

Reply:
It's an org secret enabled for specific repos and I added this one.